lib/faraday/adapter/em_synchrony.rb

```diff
@@ -26,6 +26,11 @@ def call(env)
         # Queue requests for parallel execution.
         if env[:parallel_manager]
           env[:parallel_manager].add(request, http_method, request_config(env)) do |resp|
+            if (req = env[:request]).stream_response?
```

Comment:
I'm not 100% sure of it as well, but here are my 2 cents.
I personally like having a `perform_request` method on most of the adapters. I would suggest having a `perform_stream_request` method to manage stream logics and having the `call` method calling the proper "perform" based on a check with `stream_response?`.

The common logic will therefore be something like (extremely simplified):

```ruby
def call(env)
  if req.stream_response?
    perform_stream_request
  else
    perform_request
  end
end
```

At this point, `perform_stream_request` can be implemented in the common `Faraday::Adapter` triggering the warning message and override this method only in the adapters where
you want to provide a proper implementation.
I really hope this is clear, but please do let me know if that's not the case!

Good work btw 👍!